src/api.js

```diff
@@ -280,6 +280,53 @@ api.declare({
 });
 
 api.declare({
+  name: 'badges',
+  title: 'Latest Build Status Badge',
+  description: [
+    'Checks the status of the latest build of a given branch ',
+    'and returns corresponding badge image.',
+  ].join('\n'),
+  stability: 'experimental',
+  method: 'get',
+  route: '/status/:owner/:repo/:branch',
+  output: 'status-badges.json',
+}, async function(req, res) {
+  // Extract owner, repo and branch from request into variables
+  let {owner, repo, branch} = req.params;
+
+  // Look up the installation ID in Azure. If no such owner in the table, no error thrown
+  let ownerInfo = await this.OwnersDirectory.load({owner}, true);
+
+  if (ownerInfo) {
+    try {
+      let instGithub = await this.github.getInstallationGithub(ownerInfo.installationId);
+      // Statuses is an array of status objects
+      let statuses = await instGithub.repos.getStatuses({owner, repo, ref: branch});
+      let status = statuses.find(statusObject => statusObject.context.startsWith('Taskcluster'));
+
+      // The weird switch is here for the future changes
+      switch (status.state) {
+        case 'success':
+          res.reply({status: 'success'});
```

Comment:
oh darn, just saw your reply. ok, I'll make assets/ (in the root  this one, right?)